### PR TITLE
runtime: recognize .busted files as lua filetype

### PR DESCRIPTION
**Pull request by craigmac**

### CODE REVIEW

File: src/testdir/test_filetype.vim [hunk 1]
```diff
@@ -90,6 +90,7 @@ let s:filename_checks = {
     \ 'blueprint': ['file.blp'],
     \ 'bsdl': ['file.bsd', 'file.bsdl'],
     \ 'bst': ['file.bst'],
+    \ 'busted': ['.busted'],
```
Comment (zeertzjq) [inline src/testdir/test_filetype.vim]: Shouldn't this be added to the `'lua': ` below?

**Comment by brammool** [comment]
Are these file literally called ".busted"?  So you can only have one per directory, and they are hidden files?

**Comment by craigmac** [comment]
> Are these file literally called ".busted"? So you can only have one per directory, and they are hidden files?

From what I understand in the [usage](https://lunarmodules.github.io/busted/) section, the config files are literally `.busted`, I'm presuming they are hidden (with leading '.'), and it seems you create one in the root (doesn't say whether it supports more than one per directory).

**Comment by brammool** [comment]
> > Are these file literally called ".busted"? So you can only have one
> > per directory, and they are hidden files?
> 
> From what I understand in the
> [usage](https://lunarmodules.github.io/busted/) section, the config
> files are literally `.busted`, I'm presuming they are hidden (with
> leading '.'), and it seems you create one in the root (doesn't say
> whether it supports more than one per directory).

OK, the section "Predefined Busted Tasks" explains it.

The test fails, since ".busted" gets filetype "lua", but the test
expects the filetype "busted".  So should the filetype detection be
adjusted or the test?

A quick look at the docs suggest that this .busted file has Lua syntax,
but follows test specs with a certain format.  What we often do in such
cases is use a specific filetype, and have the indent/syntax/filetype
plugin files load the more generic file type plugins.  That allows for
tuning to this specific filetype when needed.

-- 
My Go, this amn keyboar oesn't have a .

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by craigmac** [comment]
@brammool I've changed it now to just put `.busted` in the correct spot in the test file, sorry about that, it's completely lua, no extra need for distinct indent/syntax/ftplugin rules.

**Pull request closed without merging**